homeassistant/components/zha/sensor.py

```diff
@@ -1249,3 +1265,44 @@ class SonoffPresenceSenorIlluminationStatus(Sensor):
     def formatter(self, value: int) -> int | float | None:
         """Numeric pass-through formatter."""
         return SonoffIlluminationStates(value).name
+
+
+@MULTI_MATCH(cluster_handler_names=CLUSTER_HANDLER_THERMOSTAT)
+# pylint: disable-next=hass-invalid-inheritance # needs fixing
+class PiHeatingDemand(Sensor):
+    """Sensor that displays the percentage of heating power used.
+
+    Optional thermostat attribute.
+    """
+
+    _unique_id_suffix = "pi_heating_demand"
+    _attribute_name = "pi_heating_demand"
+    _attr_translation_key: str = "pi_heating_demand"
+    _attr_icon: str = "mdi:radiator"
+    _attr_native_unit_of_measurement = PERCENTAGE
+    _decimals = 0
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
```

Comment:
Should this also be a diagnostic sensor?

Reply:
I have been on the fence about that. It depends on the definition of diagnostic in this context. I did indeed write "heating power used", but the official specification says: "level of heating demanded". This information is not the primary function of the device. A door sensor would have the opening entity not as diagnostic, but the magnetic force would be. I think you're right. I changed it.

Reply:
(If so, also change to diagnostic match)

Reply:
Thanks, I forgot.